README.md

```diff
@@ -1,18 +1,40 @@
+UV-MBED
+----
+
 ## Overview
 UV-MBED is a cross-platform library allowing asynchronous TLS communication. 
 This is done by combinining [libuv](https://github.com/libuv/libuv) with [mbedTLS](https://github.com/ARMmbed/mbedtls.git)
+(see below for using other TLS implementations)
 
-### Features
+## Features
 * async TLS over TCP
+* Flexible TLS engine support
+* [pkcs#11](https://en.wikipedia.org/wiki/PKCS_11) support with default(mbedTLS) engine
 
-### API
+## API
 API is attempted to be consistent with [libuv API](http://docs.libuv.org/en/v1.x/api.html)
 
-### Suuported Platforms
+## Suuported Platforms
 * Linux
 * Darwin/MacOS
 * Windows
 
+## TLS engine support (BYFE - Bring Your Favorite Engine)
+If using mbedTLS does not work for you,
+for example you're already using another TLS library for your project, there is a way to use it inside _uv-mbed_.
+Two API [interfaces](include/uv_mbed/tls_engine.h) are defined for that purpose:
```

Comment:
consider changing [interfaces] to [interfaces are defined]? i read right past that link when looking for 'where' the interfaces were defined :(